Integer/integer.h

```diff
@@ -10,8 +10,41 @@ class Integer:
 public:
     Integer(int32_t value = 0); // Default Ctor + Ctor with value
 
+    Integer operator*(const int & a);
```

Comment:
It's good. 
One small thing - operators +-/*<> should be const:
Integer operator-(const int & a) const;

Otherwise you will have a problem when try to compare two const references.